go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -82,45 +84,140 @@ func (ovn *Controller) getLoadBalancerVIPS(
 	return raw, nil
 }
 
+// deleteLoadBalancerVIP removes the VIP as well as any reject ACLs associated to the LB
 func (ovn *Controller) deleteLoadBalancerVIP(loadBalancer, vip string) {
 	vipQuotes := fmt.Sprintf("\"%s\"", vip)
-	stdout, stderr, err := util.RunOVNNbctl("--if-exists", "remove",
-		"load_balancer", loadBalancer, "vips", vipQuotes)
+	stdout, stderr, err := util.RunOVNNbctl("--if-exists", "remove", "load_balancer", loadBalancer, "vips", vipQuotes)
 	if err != nil {
 		klog.Errorf("Error in deleting load balancer vip %s for %s"+
 			"stdout: %q, stderr: %q, error: %v",
 			vip, loadBalancer, stdout, stderr, err)
+		// if we hit an error and fail to remove load balancer, we skip removing the rejectACL
+		return
 	}
+	ovn.RemoveServiceEndpoints(loadBalancer, vip)
+	ovn.deleteLoadBalancerRejectACL(loadBalancer, vip)
+	ovn.RemoveServiceLB(loadBalancer, vip)
 }
 
-func (ovn *Controller) createLoadBalancerVIP(lb string, serviceIP string, port int32, ips []string, targetPort int32) error {
-	klog.V(5).Infof("Creating lb with %s, %s, %d, [%v], %d", lb, serviceIP, port, ips, targetPort)
-
-	// With service_ip:port as a VIP, create an entry in 'load_balancer'
-	// key is of the form "IP:port" (with quotes around)
-	key := fmt.Sprintf(`"%s"`, util.JoinHostPortInt32(serviceIP, port))
+func (ovn *Controller) configureLoadBalancer(lb, serviceIP string, port int32, endpoints string) error {
```

Comment:
@trozet let's make this function do the work of formatting the endpoints the way it needs. instead of `endpoints string` do `endpoints []string` and then you don't have to recombine the endpoints for SetServiceEndpointsToLB() too. Bonus, you can pass `nil` for an empty array rather than "".

Reply:
will do